### PR TITLE
refactor(pagination): split into smaller files for readability

### PR DESCRIPTION
**Pull request by Geobert**
Split `pagination.rs ` into smaller files for readability

### CODE REVIEW

File: src/pagination/mod.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
+use crate::cobalt_model::pagination_config::Include;
+use crate::cobalt_model::pagination_config::PaginationConfig;
+use crate::cobalt_model::permalink;
+use crate::cobalt_model::slug;
+use crate::cobalt_model::SortOrder;
+use crate::document;
+use crate::document::Document;
+use crate::error::*;
+use liquid;
+use std::cmp::Ordering;
```
Comment (epage) [inline src/pagination/mod.rs]: - `use liquid` shouldn't be needed anymore
- Please separate `use`s by std, external, internal

Comment (Geobert) [reply src/pagination/mod.rs]: Thank you for the review, it's done! :D

**Comment by epage** [comment]
Looks good, feel free to squash

**Comment by Geobert** [comment]
@epage Done! Thank you! I'm beginning on pagination on dates very soon :)

**Comment by epage** [comment]
In case you didn't notice, apparently, some things failed in rustfmt and clippy

**Comment by Geobert** [comment]
> 
> 
> In case you didn't notice, apparently, some things failed in rustfmt and clippy

It seems that clippy has changed some rules between 1.31 and 1.33, shall I update to 1.33 as well?

**Comment by epage** [comment]
> It seems that clippy has changed some rules between 1.31 and 1.33, shall I update to 1.33 as well?

If you want.

**Pull request merged**